tests/Feature/Connect/Actions/BuildClientPatchDataActionTest.php

```diff
@@ -421,7 +424,267 @@ public function testLocallyOptedOutOfSubsetsAndLoadedOptedOutSubsetHash(): void
+            publishedCount: 3,
+        );
+        $specialtyGame = $this->createGameWithAchievements(
+            $this->system,
+            'Dragon Quest III [Subset - Special]',
+            publishedCount: 2,
+        );
+
+        $this->upsertGameCoreSetAction->execute($baseGame);
+        $this->upsertGameCoreSetAction->execute($specialtyGame);
+        $this->associateAchievementSetToGameAction->execute($baseGame, $specialtyGame, AchievementSetType::Specialty, 'Special');
+
+        $specialtyGameHash = GameHash::factory()->create(['game_id' => $specialtyGame->id]);
+        $user = User::factory()->create(['websitePrefs' => self::OPT_IN_TO_ALL_SUBSETS_PREF_ENABLED]);
+
+        // Act
+        $result = (new BuildClientPatchDataAction())->execute(gameHash: $specialtyGameHash, user: $user);
+
+        // Assert
+        $this->assertTrue($result['Success']);
+
+        // ... root data should be from the base game ...
+        $this->assertEquals($baseGame->id, $result['PatchData']['ID']);
+        $this->assertEquals($baseGame->title, $result['PatchData']['Title']);
+        $this->assertEquals($baseGame->ImageIcon, $result['PatchData']['ImageIcon']);
+        $this->assertCount(3, $result['PatchData']['Achievements']); // the base game's achievements
+
+        // ... RP should be from the specialty game ...
+        $this->assertEquals($specialtyGame->RichPresencePatch, $result['PatchData']['RichPresencePatch']);
+    }
+
+    public function testItResolvesRootDataCorrectlyForExclusiveSet(): void
+    {
+        // Arrange
+        $baseGame = $this->createGameWithAchievements(
+            $this->system,
+            'Dragon Quest III',
+            publishedCount: 3,
+        );
+        $exclusiveGame = $this->createGameWithAchievements(
+            $this->system,
+            'Dragon Quest III [Subset - Exclusive]',
+            publishedCount: 2,
+        );
+
+        $this->upsertGameCoreSetAction->execute($baseGame);
+        $this->upsertGameCoreSetAction->execute($exclusiveGame);
+        $this->associateAchievementSetToGameAction->execute($baseGame, $exclusiveGame, AchievementSetType::Exclusive, 'Exclusive');
+
+        $exclusiveGameHash = GameHash::factory()->create(['game_id' => $exclusiveGame->id]);
+        $user = User::factory()->create(['websitePrefs' => self::OPT_IN_TO_ALL_SUBSETS_PREF_ENABLED]);
+
+        // Act
+        $result = (new BuildClientPatchDataAction())->execute(gameHash: $exclusiveGameHash, user: $user);
+
+        // Assert
+        $this->assertTrue($result['Success']);
+
+        // ... root data should be from the base game ...
+        $this->assertEquals($baseGame->id, $result['PatchData']['ID']);
+        $this->assertEquals($baseGame->title, $result['PatchData']['Title']);
+        $this->assertEquals($baseGame->ImageIcon, $result['PatchData']['ImageIcon']);
+        $this->assertCount(3, $result['PatchData']['Achievements']); // ... base game's achievements ...
```

Comment:
Exclusive sets load without the core achievements, so there's even less reason for the root data to be from the core game.

Reply:
In latest, `Title` and `ImageIcon` come from the base game, whereas `ID` and `Achievements` come from the subset.